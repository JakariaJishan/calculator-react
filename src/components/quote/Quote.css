blockquote {
    font-size: 24px;
    font-weight: bold;
    margin: 20px 0;
    color: rgb(63, 63, 63);
}

.quote {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 80vh;
}