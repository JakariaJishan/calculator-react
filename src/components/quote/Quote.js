import React, { useEffect, useState } from 'react';
import './Quote.css';

function Quote() {
  const [quote, setquote] = useState({});

  useEffect(() => {
    fetch('https://api.quotable.io/random')
      .then((response) => response.json())
      .then((data) => {
        setquote(data);
      });
  }, []);

  return (
    <div className="quote">
      {quote.content ? (
        <blockquote>
          &ldquo;
          {quote.content}
          &rdquo; -
          {' '}
          {quote.author}
        </blockquote>
      ) : (
        <p>Loading ....</p>
      )}
    </div>
  );
}

export default Quote;
