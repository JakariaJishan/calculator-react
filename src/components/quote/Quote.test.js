import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Quote from './Quote';

describe('Quote testing', () => {
  it('snapshot matching', () => {
    const tree = renderer.create(<BrowserRouter><Quote /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Document testing', () => {
    render(<BrowserRouter><Quote /></BrowserRouter>);
    const navbarUi = screen.getByText('Loading ....');
    expect(navbarUi).toBeInTheDocument();
  });
});
