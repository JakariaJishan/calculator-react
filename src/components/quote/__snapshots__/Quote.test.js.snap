// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote testing snapshot matching 1`] = `
<div
  className="quote"
>
  <p>
    Loading ....
  </p>
</div>
`;
